lib/symboldatabase.cpp

```diff
@@ -7089,7 +7089,14 @@ ValueType::MatchResult ValueType::matchParameter(const ValueType *call, const Va
 
 ValueType::MatchResult ValueType::matchParameter(const ValueType *call, const Variable *callVar, const Variable *funcVar)
 {
-    ValueType::MatchResult res = ValueType::matchParameter(call, funcVar->valueType());
+    ValueType vt;
+    auto pvt = funcVar->valueType();
+    if (pvt && funcVar->isArray()) {
```

Comment:
Does this work when array is passed by reference(ie `void f(const T (&array)[N])`)? Maybe we should call `matchParameter` first and then if it doesn't match adjust the valuetype if `funcVar->isArray()`?

Reply:
It works in the sense that there is no FP if I change the test case accordingly. But ValueType can't store the dimension N anyway.